logging/redaction/log_redaction.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+from typing import List
+
+from apache_beam import CombineFn, CombineGlobally, DoFn, io, ParDo, Pipeline, WindowInto
+from apache_beam.error import PipelineError
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.transforms.window import FixedWindows
+
+from google.cloud import logging_v2
+
+
+# TODO: Place inspection and de-identification configurations
+
+class PayloadAsJson(DoFn):
+    '''Convert PubSub message payload to UTF-8 and return as Json'''
+    def process(self, element):
+        yield json.loads(element.decode('utf-8'))
+
+
+class BatchPayloads(CombineFn):
+    '''Opinionated way to batch all payloads in the window'''
+
+    def create_accumulator(self):
+        return []
+
+    def add_input(self, accumulator, input):
+        accumulator.append(input)
+        return accumulator
+
+    def merge_accumulators(self, accumulators):
+        merged = []
+        for accumulator in accumulators:
+            for item in accumulator:
+                merged.append(item)
+        return merged
+
+    def extract_output(self, accumulator):
+        return accumulator
+
+
+# TODO: Placeholder for LogRedaction class
+
+
+class IngestLogs(DoFn):
+    '''Ingest payloads into destination log'''
+
+    def __init__(self, destination_log_name):
+        self.destination_log_name = destination_log_name
+        self.logger = None
+
+    def _replace_log_name(self, entry):
+        # update log name in the entry with destination log
```

Comment:
nit: python method docstrings should be `descriptive-style` (updates) rather than `imperative-style` (update).